.github/workflows/main.yml

```
@@ -0,0 +1,49 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch: 
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  markdown-lint:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@main
+      - name: lint all mardownfiles
+        run: |
+          npm install
+          node_modules/.bin/markdownlint ./posts/ --ignore node_modules --ignore v5
```

If we add scripts as suggested by my other comments, it seems like this would work?
```suggestion
          yarn install
          yarn lint
```